internal/store/interfaces.go

```diff
@@ -10,78 +10,43 @@ import (
 
 type CompareResult int
 
-// StatusTrackingGateway is an optional extension
-// of Gateway. If supported by a Store, when
-// a Gateway is synced to an external location,
-// its corresponding callbacks should
-// be called.
-// TODO Remove
-type StatusTrackingGateway interface {
-	Gateway
-
-	TrackSync(ctx context.Context, sync func() (bool, error)) error
-}
-
 // Gateway describes a gateway.
 type Gateway interface {
-	ID() core.GatewayID
-	Bind(ctx context.Context, route Route) []string
-	Remove(ctx context.Context, id string) error
-	Resolve() core.ResolvedGateway
-	CanFetchSecrets(secrets []string) (bool, error)
-}
-
-type NewGateway interface {
 	ID() core.GatewayID
 	Resolve() core.ResolvedGateway
 	CanFetchSecrets(secrets []string) (bool, error)
 }
 
-// StatusTrackingRoute is an optional extension
-// of Route. If supported by a Store, when
-// a Route is bound or fails to be bound to
-// a gateway, its corresponding callbacks should
-// be called. At the end of any methods affecting
-// the route's binding, SyncStatus should be called.
-type StatusTrackingRoute interface {
-	Route
-
-	SyncStatus(ctx context.Context) error
-	OnGatewayRemoved(gateway Gateway)
-}
-
 // Route should be implemented by all route
 // source integrations
 type Route interface {
 	ID() string
 }
 
-// Store is used for persisting and querying gateways and routes
-// TODO Remove
-type Store interface {
+type ReadStore interface {
```

Comment:
I separated into composable `ReadStore` and `WriteStore` interfaces. This helped me reason with where we're sending the store in our program entrypoint (envoy middleware, for example, doesn't need to update the store at all - only read it)